pexpect/pxssh.py

```diff
@@ -21,389 +21,234 @@
-                       cwd=cwd, env=env, ignore_sighup=ignore_sighup, echo=echo,
-                       encoding=encoding, codec_errors=codec_errors)
-
-        self.name = '<pxssh>'
-
-        #SUBTLE HACK ALERT! Note that the command that SETS the prompt uses a
-        #slightly different string than the regular expression to match it. This
-        #is because when you set the prompt the command will echo back, but we
-        #don't want to match the echoed command. So if we make the set command
-        #slightly different than the regex we eliminate the problem. To make the
-        #set command different we add a backslash in front of $. The $ doesn't
-        #need to be escaped, but it doesn't hurt and serves to make the set
-        #prompt command different than the regex.
-
-        # used to match the command-line prompt
-        self.UNIQUE_PROMPT = "\[PEXPECT\][\$\#] "
-        self.PROMPT = self.UNIQUE_PROMPT
 
-        # used to set shell command-line prompt to UNIQUE_PROMPT.
-        self.PROMPT_SET_SH = "PS1='[PEXPECT]\$ '"
-        self.PROMPT_SET_CSH = "set prompt='[PEXPECT]\$ '"
-        self.SSH_OPTS = ("-o'RSAAuthentication=no'"
-                + " -o 'PubkeyAuthentication=no'")
-# Disabling host key checking, makes you vulnerable to MITM attacks.
-#                + " -o 'StrictHostKeyChecking=no'"
-#                + " -o 'UserKnownHostsFile /dev/null' ")
-        # Disabling X11 forwarding gets rid of the annoying SSH_ASKPASS from
-        # displaying a GUI password dialog. I have not figured out how to
-        # disable only SSH_ASKPASS without also disabling X11 forwarding.
-        # Unsetting SSH_ASKPASS on the remote side doesn't disable it! Annoying!
-        #self.SSH_OPTS = "-x -o'RSAAuthentication=no' -o 'PubkeyAuthentication=no'"
-        self.force_password = False
+class ExceptionPxssh(ExceptionPexpect):
+    pass
+
+
+class pxssh(spawn):
+
+    _PROMPT_ESCAPES =["$", "(", ")", "[", "]"]
+    _REGEX_STRINGS = {
+        "accept_key": "(?i)are you sure you want to continue connecting",
+        "password_check": "(?i)(?:password)|(?:passphrase for key)",
+        "denied": "(?i)permission denied",
+        "terminal_type": "(?i)terminal type",
+        "closed": "(?i)connection closed by remote host",
+        "stopped": "(?i)there are stopped jobs"
+    }
+    _SSH_COMMAND = 'ssh {0} -l {1} {2}'
+    _SSH_OPTIONS = {
+        "quiet": " -q",
+        "check_local_ip": " -o'NoHostAuthenticationForLocalhost=yes'",
+        "port": " -p {0}",
+        "ssh_key": " -i {0}"
+    }
+    PROMPT_SET_SH = "PS1='[PEXPECT]\$ '"
+    PROMPT_SET_CSH = "set prompt='[PEXPECT]\$ '"
+    UNIQUE_PROMPT = "\[PEXPECT\][\$\#] "
+
+    def _build_ssh_command(self):
+        """
+            Method to build the ssh command string.
+        :return:
+        """
+        options = ''.join([" -o '%s=%s'" % (o, v) for (o, v) in self.options.items()])
+
+        for key, val in pxssh._SSH_OPTIONS.iteritems():
```

Comment:
`.iteritems()` is gone in Python 3 (`.items()` returns an iterator anyway). It's possible to use a wrapper function that calls `.iteritems()` on Py2, but as there's only 4 things in the dict anyway, let's just use `.items()` on it.